### PR TITLE
Implement a "familiar" name format

### PR DESCRIPTION
**Pull request by ehoogerbeets**
This is as familiar as you can get and yet not be rude to a stranger for each locale.

### Checklist

* [x] At least one test case is included for this feature or bug fix.
* [x] `ReleaseNotes` has added or is not needed.
* [x] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [x] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

File: js/data/locale/ko/name.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
 	"components": {
 		"formal_short": "pgf",
-		"formal_long": "pgf"
+		"formal_long": "pgf",
+        "familiar": "pgf"
```
Comment (laughnan) [inline js/data/locale/ko/name.json]: `[nit]` tab formatting here

File: .gitignore [hunk 2]
```diff
@@ -23,3 +23,4 @@ qt/FileReader/release/
 qt/output/
 
 /node_modules/
+/dist/
```
Comment (laughnan) [inline .gitignore]: did you mean to include this?

Comment (ehoogerbeets) [reply .gitignore]: yes, that is the destination for the build script when you build the "distribution"

File: js/test/name/testnamefmt.js [hunk 3]
```diff
@@ -693,7 +730,25 @@ module.exports.testnamefmt = {
         test.equal(fmt.format(name), "닥터 박은성");
         test.done();
     },
-    
+
+    testNameFmtKOFormalLong: function(test) {
```
Comment (ehoogerbeets) [inline js/test/name/testnamefmt.js]: Name conflict with the previous function

File: js/test/name/testnamefmt.js [hunk 4]
```diff
@@ -745,9 +746,25 @@ module.exports.testnamefmt = {
...
+            locale: "ko-KR"
+        });
+        var fmt = new NameFmt({
+            style: "familiar",
+            locale: "ko-KR"
+        });
+
+        test.equal(fmt.format(name), "박은성 님");
+        test.done();
+    },*/
```
Comment (gouniLee) [inline js/test/name/testnamefmt.js]: @ehoogerbeets What do you think about the set default value even developer don't put suffix? (This testcase doesn't activate at this point)

**Comment by gouniLee** [comment]
@ehoogerbeets Regarding the Korean case, I've updated the component to include `suffix` It's pretty important to put 'suffix` especially not to be rude. So I want to get your opinion to give default `suffix` value.

File: js/test/name/testnamefmt.js (hunk 4 above)
Comment (ehoogerbeets) [reply js/test/name/testnamefmt.js]: We can do that in a separate PR

**Pull request merged**